### DISCUSSION TITLE
Anyone have a HD-PVR2 working in SageTV on unRAID?

### DISCUSSION THREAD

**Discussion by wayner9** [OpenSageTV/Community#139] [General]
I have been using HD-PVRs with component input for about 15 years, but I recently had to switch to Xfinity/Ignite cable boxes that don't have component outputs.  So now I need to capture an HDMI signal.  I have tried a couple of different HDMI-component adapters but they degraded the video quality.  I bought a HD-PVR2 on ebay and tried using it, but it didn't work on my first try.  I didn't play around too much as I am not sure if they are supported.  Anyone know if a HD-PVR2 should work in unRAID?

I think this is really a question of whether the DVB Driver plugin for unRAID supports the HD-PVR2 since that is the plugin that has the drivers for capture devices for unRAID.

**Comment by emveepee** [comment]
Unlike the HDPVR the HDVPR2 has no built Linux drivers or kernel support so it won't work out of the box.   The best you could hope for being able to compile this https://github.com/jpoet/HauppaugeUSB and be able to run a command line capture.  The good news is on Debian and Ubuntu I have found it very stable with the 1212 eventually I would have to reboot the box when I noticed poor image quality.

You should note that not all HDPVR2 devices are supported.

A DVB plugin would not help for the HDVPR or HDPVR2

**Reply by wayner9** [reply]
The DVB plugin definitely helps with the HD-PVR (and other tuning devices) on unRAID.  See: https://forums.sagetv.com/forums/showthread_t_64737.html?t=64737 and this 2016 thread by stuckless on the unRAID forums:  https://forums.unraid.net/topic/46830-setting-up-unraid-docker-for-sagetv-server-but-need-hdpvr-access/

At this time you needed something called the unRAID DVB edition which was an alternate kernel (not sure if that is the right terminology) - I have this on my older server which is still running unRAID version 6.4.1 and has been live since about 2016.  When you used this method you didn't have to install any drivers, just the alternate kernel.

This changed somewhere around unRAID version 6.10 and became a regular unRAID plugin.  I have been running the new plugin on my newer unRAID server which I have been using since about 2021.  The plugin adds drivers for various tuning devices.

Here is a short thread from the old forums with Striker:WG and I discussing this issue  https://forums.sagetv.com/forums/showthread_t_66671.html?t=66671

The thread on this plugin on the unRAID forum is here:  https://forums.sagetv.com/forums/showthread_t_66671.html?t=66671

But to get back to the main issue, I may have to use the HDPVR2 on a Windows PC and then use it as a network encoder.  Or go with other HDMI tuning devices.  I did buy one of those ~$25 HDMI-USB capture devices but I haven't spent much time trying to get it work with OpenDCT.  I will have to get around to that to see how it does.  Or go to a uRAY type device.

**Reply by paulcheck** [reply]
> Unlike the HDPVR the HDVPR2 has no built Linux drivers or kernel support so it won't work out of the box. The best you could hope for being able to compile this https://github.com/jpoet/HauppaugeUSB and be able to run a command line capture. The good news is on Debian and Ubuntu I have found it very stable with the 1212 eventually I would have to reboot the box when I noticed poor image quality.
> 
> You should note that not all HDPVR2 devices are supported.
> 
> A DVB plugin would not help for the HDVPR or HDPVR2

I just replaced my old HD PVR 1212 with a new HD PVR 2 1512.  I have it almost working using the HauppaugeUSB git code. The issue is that the audio is low. Of course I tried the boost option but no effect.  Actually, not only is the audio low, but it already seems "boosted" in the sense that I can hear the slight hiss or noise that comes with gain. Has anyone else has this issue?

**Reply by wayner9** [reply]
Which OS - Windows, un RAID or some other OS?

**Comment by emveepee** [comment]
It possible that the DVB plugin also installs the v4l2 subsystem, I don't know how basic the unRaid core is.
